git-restore: add page (#3247)
tldr-pages/tldr: pages/common/git-restore.md

# git restore

> Restore working tree files. Requires git version 2.23+.
> See also `git checkout`.
> More information: <https://git-scm.com/docs/git-restore/>.

> - Restore a file that was accidentally deleted:

Reviewer: I don't think the "accidentally" is needed here :)

Also, maybe specify that this fetches the file contents from the current commit / HEAD?

> - Restore a file from the current HEAD that was previously deleted:

Reviewer: How about something like this?

```suggestion
- Restore a deleted file from the contents of the current commit (HEAD):
```

Author: Sure!

`git restore {{path/to/file}}`

> - Restore a file to an earlier version:

Reviewer: The other version is not necessarily earlier. It could simply be from a different branch. Maybe say that explicitly.

`git restore --source {{commit}} {{path/to/file}}`

> - Restore all the files in the current working tree to the last committed version:

Reviewer: Alternatively:

> Undo any uncommitted changes to tracked files.

I'm not sure which version I prefer. Yours is certainly more explicit. I'll wait to see what others have to say :)

Reviewer 2: Does this use `HEAD` as "last committed version"? Too bad I don't have the latest Git version so I cannot test it out. If this uses `HEAD`, saying "the last committed version" could be unclear and I would suggest using @waldyrious' version.

Reviewer 3: I prefer the shorter version. It succinctly explains what the command does without losing any information.

Reviewer: > Does this use HEAD as "last committed version"?

Yes, just tested — it defaults to HEAD if no revision is specified (what else would it do, anyway?).

> If this uses HEAD, saying "the last committed version" could be unclear

Well, true, but for most cases this will hold. But yeah, if you've checked out, say, a tag, that typically won't be the latest commit in the branch. Granted, one usually won't make changes in a non-latest commit of a branch, but it's certainly possible, so I see your point.

Reviewer 2: Yeah I'm totally nitpicking here, but whatever, that was a long way to say I prefer the shortest version 😆

Author: I've taken the shorter one, but added a bit to try and make it clearer what it is restoring to. Does this work?

Reviewer: @sblr why ask a question then mark the conversation as resolved? :P

But yes, I agree with your changes. :+1:
